File: DigitalLicenseApplication/Backend/DigitalLicenseApplication/DigitalLicenseApplication/Service/Formservice.cs
using DigitalLicenseApplication.Interface.Repository;
using DigitalLicenseApplication.Interface.Service;

namespace DigitalLicenseApplication.Service
{
    public class Formservice:IFormservice
    {
        private readonly IFormrepo _formrepo;
        public Formservice(IFormrepo formrepo)
        {
            _formrepo = formrepo;
        }
        public IEnumerable<Models.Form> GetAllForms()
        {
            return _formrepo.GetAllForms();
        }
        public Models.Form GetFormById(int id)
        {
            return _formrepo.GetFormById(id);
        }
        public IEnumerable<Models.Form> GetFormsByUserId(int userId)
        {
            return _formrepo.GetFormsByUserId(userId);
        }
        public IEnumerable<Models.Form> GetFormsByStatus(string status)
        {
            return _formrepo.GetFormsByStatus(status);
        }
        public void SubmitForm(Models.Form form)
        {
            _formrepo.SubmitForm(form);
        }
        public void UpdateForm(Models.Form form)
        {
            _formrepo.UpdateForm(form);
        }
        public void DeleteForm(int id)
        {
            _formrepo.DeleteForm(id);
        }
        public IEnumerable<Models.Form> GetFormsByDepartment(int departmentId)
        {
            return _formrepo.GetFormsByDepartment(departmentId);
        }
    }
}

File: DigitalLicenseApplication/Frontend/src/app/services/form/formservice.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Forms } from '../../models/form.model';

@Injectable({
  providedIn: 'root'
})
export class Formservice {
  private apiUrl = 'https://localhost:7034/api/Form';

  constructor(private http: HttpClient) {}

  getAllForms(): Observable<Forms[]> {
    return this.http.get<Forms[]>(this.apiUrl);
  }

  getFormById(id: number): Observable<Forms> {
    return this.http.get<Forms>(`${this.apiUrl}/${id}`);
  }

  addForm(form: any): Observable<Forms> {
    return this.http.post<Forms>(this.apiUrl, form);
  }

  updateForm(formId: number, form: any): Observable<any> {
    return this.http.put(`${this.apiUrl}/${formId}`, form);
  }

  deleteForm(formId: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${formId}`);
  }

  getFormsByDepartment(departmentId: number): Observable<Forms[]> {
    return this.http.get<Forms[]>(`${this.apiUrl}/department/${departmentId}`);
  }
}
